### PR TITLE
Add missing GITHUB_TOKEN to CI

### PR DESCRIPTION
**Pull request by khos2ow**
Pass [`GITHUB_TOKEN` secret](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#about-the-github_token-secret) (injected by GitHub) as an environment variable to the CI step.

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+      - name: Set up Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16.x
+
+      - name: Generate lists
+        run: |
+          go run . -count 10
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
Comment (khos2ow) [inline .github/workflows/test.yaml]: @mvdan I created this job solely as an example to make sure it works, which it does and the result can be viewed [here](https://github.com/mvdan/corpus/runs/2548008823#step:4:13). Let me know to revert back and delete `test.yaml` once you reviewed this PR.

Comment (mvdan) [reply .github/workflows/test.yaml]: SGTM. We could leave the test in too, just to check that a very simple `go run` works.

Comment (khos2ow) [reply .github/workflows/test.yaml]: Or actually on the second thought the `test.yaml` can remain as is to validate the state of PR and make sure `go run` still works.

Comment (khos2ow) [reply .github/workflows/test.yaml]: Haha, yeah exactly what you said too! Let me rephrase the commit message then and this should be good to go.

**Review by mvdan — APPROVED** [review]
